File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/library/items/media/audio/AudioCacheConfiguration.java
package com.lasthopesoftware.bluewater.client.browsing.library.items.media.audio;

import com.lasthopesoftware.bluewater.client.browsing.library.items.media.files.cached.configuration.IDiskFileCacheConfiguration;
import com.lasthopesoftware.bluewater.client.browsing.library.repository.Library;

import org.joda.time.Duration;


public class AudioCacheConfiguration implements IDiskFileCacheConfiguration {
	private final Library library;
	private static final String musicCacheName = "music";
	private static final long maxFileCacheSize = 2L * 1024L * 1024L * 1024L; // ~2GB

	public AudioCacheConfiguration(Library library) {
		this.library = library;
	}

	@Override
	public String getCacheName() {
		return musicCacheName;
	}

	@Override
	public Library getLibrary() {
		return library;
	}

	@Override
	public long getMaxSize() {
		return maxFileCacheSize;
	}

	@Override
	public Duration getCacheItemLifetime() {
		return null;
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/playback/caching/AudioCacheConfiguration.kt
package com.lasthopesoftware.bluewater.client.playback.caching

import com.lasthopesoftware.bluewater.client.browsing.files.cached.configuration.IDiskFileCacheConfiguration
import org.joda.time.Duration

object AudioCacheConfiguration : IDiskFileCacheConfiguration {
	private const val musicCacheName = "music"
	private const val maxFileCacheSize = 1024L * 1024L * 1024L // ~1GB

	override val cacheName: String = musicCacheName

	override val maxSize: Long = maxFileCacheSize

	override val cacheItemLifetime: Duration? = null
}
